sec_certs/sample/certificate.py

```diff
@@ -56,19 +76,22 @@ def __eq__(self, other: object) -> bool:
             return False
         return self.dgst == other.dgst
 
-    def to_dict(self) -> Dict[str, Any]:
+    def __hash__(self) -> int:
+        return hash(self.dgst)
```

Comment:
Hm, this is problematic then. We want two things:

- Have `Certificate` objects comparable using `==` operator (so that, e.g., dictionaries carrying `Certificate`s can be compared as well, which assures that `Dataset` objects can be compared)
- Have `Certificate` objects *hashable* so that we can use them in sets. 

Since the classes define mutable attributes, there's a direct conflict between these two requirements?

Reply:
Yes, there is a conflict. But the second thing is in conflict with the class being mutable by itself, the conflict is not between the two requirements.

Reply:
Ok, so only `FIPSCertificate` needed to be hashable and it no longer does. So we're good.